projects/vdk-plugins/vdk-control-api-auth/src/vdk/plugin/control_api_auth/base_auth.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import json
+import logging
+import time
+from typing import Optional
+from typing import Union
+
+from requests import post
+from requests.auth import HTTPBasicAuth
+from requests_oauthlib import OAuth2Session
+from vdk.plugin.control_api_auth.auth_config import AuthConfig
+from vdk.plugin.control_api_auth.auth_config import AuthConfigFolder
+from vdk.plugin.control_api_auth.auth_config import InMemAuthConfiguration
+from vdk.plugin.control_api_auth.auth_exception import VDKAuthException
+from vdk.plugin.control_api_auth.auth_request_values import AuthRequestValues
+from vdk.plugin.control_api_auth.login_types import LoginTypes
+
```

Comment:
you mention pkce within the PR description, and I don't find any such related changes?
so `auth_pkce.py` from vdk-control-cli remains?

Reply:
Just updated the PR description. The pkce authentication flow will be added as part of a separate PR. I didn't want to put everything in a single PR, as the change would be really hard to review properly, so I opted for a multi-PR change.
The release step in the CI/CD will not be enabled until the work on the library is complete.